packages/components/rating/src/Rating.test.tsx

```diff
@@ -1,12 +1,127 @@
+import { fireEvent } from '@testing-library/dom'
 import { render, screen } from '@testing-library/react'
-import { describe, expect, it } from 'vitest'
+import userEvent from '@testing-library/user-event'
+import { useState } from 'react'
+import { describe, expect, it, vi } from 'vitest'
 
 import { Rating } from './Rating'
 
+const utils = {
+  getInput: () =>
+    screen.getByRole('slider', {
+      name: /rating/i,
+    }),
+  getStars: (container: HTMLElement) =>
+    Array.from(container.querySelectorAll('[data-part="star"]')),
+}
+
 describe('Rating', () => {
   it('should render', () => {
-    render(<Rating />)
+    const { container } = render(<Rating defaultValue={2} />)
+
+    expect(container.querySelector('[data-spark-component=rating]')).toBeInTheDocument()
+    expect(container.querySelectorAll('[data-part="star"]').length).toBe(5)
+  })
+
+  it('should be possible to interact', async () => {
+    const user = userEvent.setup()
+
+    const { container } = render(<Rating size="lg" defaultValue={1} />)
+
+    const stars = utils.getStars(container)
+    const input = utils.getInput()
+    const firstStar = stars.at(0)
+    const thirdStar = stars.at(2)
+
+    if (!thirdStar || !firstStar) throw new Error('Stars element not found')
+
+    // initial value
+    expect(input).toHaveValue('1')
+
+    await user.click(thirdStar)
+    expect(input).toHaveValue('3')
+
+    await user.click(firstStar)
+    expect(input).toHaveValue('1')
+
+    // triggering a change event by directly interacting
+    // with the HTML input (using the keyboard for instance)
+    fireEvent.change(input, { target: { value: '5' } })
+    expect(input).toHaveValue('5')
+  })
+
+  it('should handle controlled mode', async () => {
+    function ControlledRating({ onValueChange }: { onValueChange: (value: number) => void }) {
+      const [value, setValue] = useState(2)
+
+      const handleInteraction = (value: number): void => {
+        onValueChange(value)
+        setValue(value)
+      }
+
+      return <Rating value={value} onValueChange={handleInteraction} />
+    }
+
+    const onValueChangeSpy = vi.fn()
+    const user = userEvent.setup()
+    const { container } = render(<ControlledRating onValueChange={onValueChangeSpy} />)
+
+    const stars = utils.getStars(container)
+    const input = utils.getInput()
+    const fifthStar = stars.at(4)
+
+    if (!fifthStar) throw new Error('Stars element not found')
+
+    // initial value
+    expect(input).toHaveValue('2')
+
+    await user.click(fifthStar)
+    expect(input).toHaveValue('5')
+    expect(onValueChangeSpy).toHaveBeenNthCalledWith(1, 5)
```

Comment:
Why not simply:
```suggestion
    expect(onValueChangeSpy).toHaveBeenCalledWith(5)
```
?

Reply:
> To make sure we don't call the onValueChange prop more than we need to I guess
Then, I'm afraid there could be a misunderstanding here, as the `toHaveBeenNthCalledWith(a, b)` method test that the spied callback is  called with `b` value on `a` call.
If I understand correctly, you should here write something like:
```tsx
expect(onValueChangeSpy).toHaveBeenCalledTimes(1)
expect(onValueChangeSpy).toHaveBeenCalledWith(5)
```
Wdyt?